seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java

```diff
@@ -0,0 +1,309 @@
+            initialize();
+        }
+        return fileSystem.open(new Path(filePath));
+    }
+
+    @SneakyThrows
+    public <T> T doWithHadoopAuth(HadoopLoginFactory.LoginFunction<T> loginFunction) {
+        if (configuration == null) {
+            this.configuration = createConfiguration();
+        }
+        if (enableKerberos()) {
+            configuration.set("hadoop.security.authentication", "kerberos");
+            return HadoopLoginFactory.loginWithKerberos(
+                    configuration,
+                    hadoopConf.getKrb5Path(),
+                    hadoopConf.getKerberosPrincipal(),
+                    hadoopConf.getKerberosKeytabPath(),
+                    loginFunction);
+        }
+        if (enableRemoteUser()) {
+            return HadoopLoginFactory.loginWithRemoteUser(
+                    configuration, hadoopConf.getRemoteUser(), loginFunction);
+        }
+        return loginFunction.run(configuration, UserGroupInformation.getCurrentUser());
+    }
+
+    @Override
+    public void close() throws IOException {
+        synchronized (UserGroupInformation.class) {
+            try (FileSystem closedFileSystem = fileSystem) {
+                if (userGroupInformation != null) {
+                    userGroupInformation.logoutUserFromKeytab();
+                }
+            }
+        }
+    }
+
+    @SneakyThrows
+    private void initialize() {
+        this.configuration = createConfiguration();
+        if (enableKerberos()) {
+            configuration.set("hadoop.security.authentication", "kerberos");
+            initializeWithKerberosLogin();
+            return;
+        }
+        if (enableRemoteUser()) {
+            initializeWithRemoteUserLogin();
+            return;
+        }
+        this.fileSystem = FileSystem.get(configuration);
+        fileSystem.setWriteChecksum(false);
+    }
+
+    private Configuration createConfiguration() {
+        Configuration configuration = new Configuration();
+        configuration.setBoolean(READ_INT96_AS_FIXED, true);
+        configuration.setBoolean(WRITE_FIXED_AS_INT96, true);
+        configuration.setBoolean(ADD_LIST_ELEMENT_RECORDS, false);
+        configuration.setBoolean(WRITE_OLD_LIST_STRUCTURE, true);
+        configuration.set(CommonConfigurationKeys.FS_DEFAULT_NAME_KEY, hadoopConf.getHdfsNameKey());
+        configuration.set(
+                String.format("fs.%s.impl", hadoopConf.getSchema()), hadoopConf.getFsHdfsImpl());
+        hadoopConf.setExtraOptionsForConfiguration(configuration);
+        return configuration;
+    }
+
+    private boolean enableKerberos() {
+        boolean kerberosPrincipalEmpty = StringUtils.isEmpty(hadoopConf.getKerberosPrincipal());
+        boolean kerberosKeytabPathEmpty = StringUtils.isEmpty(hadoopConf.getKerberosKeytabPath());
+        boolean krb5FilePathEmpty = StringUtils.isEmpty(hadoopConf.getKrb5Path());
+        if (kerberosKeytabPathEmpty && kerberosPrincipalEmpty && krb5FilePathEmpty) {
```

Comment:
krb5FilePathEmpty is it necessarily FALSE,i think he has a default value?

Reply:
Yes, we have default value in options.

Reply:
We will set the value to `/etc/krb5.conf` if user doesn't set this value in conf, but if you want to use `HadoopFileSystemProxy` in code, there is no default value.

Reply:
ok